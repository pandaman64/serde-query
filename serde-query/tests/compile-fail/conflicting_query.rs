#[derive(serde_query::Deserialize)]
struct A {
    #[query(r#".foo.x"#)]
    field_access: String,
    #[query(r#".foo.[0]"#)]
    index_access: String,
}

#[derive(serde_query::Deserialize)]
struct B {
    #[query(r#".foo"#)]
    expect_value: String,
    #[query(r#".foo.bar"#)]
    expect_struct: String,
}

fn assert_deserialize<'de, D: serde::Deserialize<'de>>() {}

fn main() {
    // ensure that fallback implemenations work
    assert_deserialize::<A>();
    assert_deserialize::<B>();
}
