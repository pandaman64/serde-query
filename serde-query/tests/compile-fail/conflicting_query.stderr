error: Conflicting query at '.foo'. The query for field 'field_access' expects a struct while the query for field 'index_access' expects a sequence.
 --> tests/compile-fail/conflicting_query.rs:4:5
  |
4 |     field_access: String,
  |     ^^^^^^^^^^^^

error: Conflicting query at '.foo'. The query for field 'expect_value' expects a value here while the query for field 'expect_struct' expects a struct.
  --> tests/compile-fail/conflicting_query.rs:12:5
   |
12 |     expect_value: String,
   |     ^^^^^^^^^^^^
